### DISCUSSION TITLE
Is spaces optional?

### DISCUSSION THREAD

**Discussion by lsshawn** [digitalocean/supabase-on-do#27] [Q&A] [answered]
I'm not using supabase to store any images. I'm only using the database and auth.

So do I need to pay for spaces to self host supabase?

Or can I just pay $6 for a droplet?

**Comment by timothymamo** [comment, marked as answer]
Hi @lsshawn,

Spaces is optional and you can store your data on the Droplet (this obviously would mean that you would loose any data stored if the Droplet is restarted, but you seem to be ok with this).

You would need to modify the docker-compose file to specify not to use the Spaces bucket within the [storage container](https://github.com/digitalocean/supabase-on-do/blob/main/packer/supabase/docker-compose.yml#L167) (see example in [supabase repo](https://github.com/supabase/supabase/blob/master/docker/docker-compose.yml#L169)) and you would have to remove any reference of creating the Spaces bucket within the terraform files [here](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/storage.tf#L1-L35), [here](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/storage.tf#L1-L35) and [here](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/variables.tf#L9-L22), as well as the env file [here](https://github.com/digitalocean/supabase-on-do/blob/main/terraform/files/.env.tftpl#L28-L36) (I think I got them all but you'd have to check).

You will still need to attach the [Volume](https://www.digitalocean.com/products/block-storage) to the Droplet to ensure that the data being stored on the database is not lost should the droplet be restarted or upgraded.

Tim
